### ISSUE TITLE
Second segfault in solidity::langutil::SourceLocation

### ISSUE THREAD

**Issue by Changochen** [ethereum/solidity#9441]
## Description
Consider the following POC:
```solidity
contract a {
    function b () {
        mapping ( uint160 => uint [ 8 ** 30 ] ) storage b   ;
        var b = b ;
    }
}
```

Run with `solc --bin -o poc poc.sol` with an solc built with asan and we got 
```
ASAN:DEADLYSIGNAL
=================================================================
==226770==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000018 (pc 0x5595eb083f57 bp 0x7ffd60266db0 sp 0x7ffd60266da0 T0)
==226770==The signal is caused by a READ memory access.
==226770==Hint: address points to the zero page.
    #0 0x5595eb083f56 in solidity::langutil::SourceLocation::SourceLocation(solidity::langutil::SourceLocation const&) (/home/yongheng/solidity/asan/solc/solc+0x4f4f56)
    #1 0x5595eb4db2ef in boost::error_info<solidity::langutil::tag_sourceLocation, solidity::langutil::SourceLocation>::error_info(solidity::langutil::SourceLocation const&) (/home/yongheng/solidity/asan/solc/solc+0x94c2ef)
    #2 0x5595ebfa727f in solidity::langutil::ErrorReporter::error(solidity::langutil::ErrorId, solidity::langutil::Error::Type, solidity::langutil::SourceLocation const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/home/yongheng/solidity/asan/solc/solc+0x141827f)
    #3 0x5595ebfa70b0 in solidity::langutil::ErrorReporter::warning(solidity::langutil::ErrorId, solidity::langutil::SourceLocation const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/home/yongheng/solidity/asan/solc/solc+0x14180b0)
    #4 0x5595eb370354 in solidity::frontend::TypeChecker::visit(solidity::frontend::VariableDeclaration const&) (/home/yongheng/solidity/asan/solc/solc+0x7e1354)
    #5 0x5595eb3e983a in solidity::frontend::VariableDeclaration::accept(solidity::frontend::ASTConstVisitor&) const (/home/yongheng/solidity/asan/solc/solc+0x85a83a)
    #6 0x5595eb384769 in solidity::frontend::TypeChecker::visit(solidity::frontend::VariableDeclarationStatement const&) (/home/yongheng/solidity/asan/solc/solc+0x7f5769)
    #7 0x5595eb3eddda in solidity::frontend::VariableDeclarationStatement::accept(solidity::frontend::ASTConstVisitor&) const (/home/yongheng/solidity/asan/solc/solc+0x85edda)
    #8 0x5595eb403550 in void solidity::frontend::ASTNode::listAccept<std::shared_ptr<solidity::frontend::Statement> >(std::vector<std::shared_ptr<solidity::frontend::Statement>, std::allocator<std::shared_ptr<solidity::frontend::Statement> > > const&, solidity::frontend::ASTConstVisitor&) (/home/yongheng/solidity/asan/solc/solc+0x874550)
    #9 0x5595eb3eb6bb in solidity::frontend::Block::accept(solidity::frontend::ASTConstVisitor&) const (/home/yongheng/solidity/asan/solc/solc+0x85c6bb)
    #10 0x5595eb36bdd1 in solidity::frontend::TypeChecker::visit(solidity::frontend::FunctionDefinition const&) (/home/yongheng/solidity/asan/solc/solc+0x7dcdd1)
    #11 0x5595eb3e9296 in solidity::frontend::FunctionDefinition::accept(solidity::frontend::ASTConstVisitor&) const (/home/yongheng/solidity/asan/solc/solc+0x85a296)
    #12 0x5595eb362bab in solidity::frontend::TypeChecker::visit(solidity::frontend::ContractDefinition const&) (/home/yongheng/solidity/asan/solc/solc+0x7d3bab)
    #13 0x5595eb3e7e36 in solidity::frontend::ContractDefinition::accept(solidity::frontend::ASTConstVisitor&) const (/home/yongheng/solidity/asan/solc/solc+0x858e36)
    #14 0x5595eb401b76 in void solidity::frontend::ASTNode::listAccept<std::shared_ptr<solidity::frontend::ASTNode> >(std::vector<std::shared_ptr<solidity::frontend::ASTNode>, std::allocator<std::shared_ptr<solidity::frontend::ASTNode> > > const&, solidity::frontend::ASTConstVisitor&) (/home/yongheng/solidity/asan/solc/solc+0x872b76)
    #15 0x5595eb3e76df in solidity::frontend::SourceUnit::accept(solidity::frontend::ASTConstVisitor&) const (/home/yongheng/solidity/asan/solc/solc+0x8586df)
    #16 0x5595eb361ddf in solidity::frontend::TypeChecker::checkTypeRequirements(solidity::frontend::SourceUnit const&) (/home/yongheng/solidity/asan/solc/solc+0x7d2ddf)
    #17 0x5595eaff9bf1 in solidity::frontend::CompilerStack::analyze() (/home/yongheng/solidity/asan/solc/solc+0x46abf1)
    #18 0x5595eaffbe68 in solidity::frontend::CompilerStack::parseAndAnalyze() (/home/yongheng/solidity/asan/solc/solc+0x46ce68)
    #19 0x5595eaffcbbc in solidity::frontend::CompilerStack::compile() (/home/yongheng/solidity/asan/solc/solc+0x46dbbc)
    #20 0x5595eada2d03 in solidity::frontend::CommandLineInterface::processInput() (/home/yongheng/solidity/asan/solc/solc+0x213d03)
    #21 0x5595eae474ce in main (/home/yongheng/solidity/asan/solc/solc+0x2b84ce)
    #22 0x7fc34a172b96 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b96)
    #23 0x5595ead8b2c9 in _start (/home/yongheng/solidity/asan/solc/solc+0x1fc2c9)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/home/yongheng/solidity/asan/solc/solc+0x4f4f56) in solidity::langutil::SourceLocation::SourceLocation(solidity::langutil::SourceLocation const&)
==226770==ABORTING
```
## Environment

- Compiler version:Version: 0.6.12-develop.2020.7.17+commit.d808302c.Linux.g++
- Target EVM version (as per compiler settings):
- Framework/IDE (e.g. Truffle or Remix):
- EVM execution environment / backend / blockchain client:
- Operating system: Ubuntu 16

**Comment by chriseth** [comment]
This is very likely solved by removing `var` in #9428

**Comment by bshastry** [comment]
May be related to https://github.com/ethereum/solidity/issues/9389

**Comment by chriseth** [comment]
Closing as fixed by #9428 on the breaking branch.
